### PR TITLE
Color transfer

### PR DESCRIPTION
**Pull request by MichielStock**

### CODE REVIEW

**Comment by MichielStock** [comment]
Starting to look good.

- add your name and a title to the notebook
- maybe there is a more elegant way of downloading the images? You might check some of the pluto docs. Would be nice if there are buttons I can select my own images.
- best to split your functions into different cells
- you can represent every pixel as the RGB + location and use k-means clustering to get prototype pixels. The obtained centres can be mapped and used for prediction again. This way you don't need downscaling.
- When making your own types (`Cluster`) adding type annotation greatly improves performance
- you might tidy up the organization of your functions.

Bottom line: make sure you don't have to do OT on all the pixels. Make this part of your preprocessing.

This was my attempt (a long time ago): https://github.com/MichielStock/Teaching/blob/master/Optimal_transport/color_transfer.py

Below is the advice I got from Mathieu Blondel

Pasted image 20210505131036

Good luck!

**Comment by juhlee** [comment]
Thank you for the detailed comment!

**Comment by MichielStock** [comment]
It is shaping up to be a *very* impressive project. When uploading your notebook, I still have some trouble doing this right, mainly because the links to images are relative (better use hyperlinks to github). Maybe have someone else testing it that works locally fine.

image

**Comment by juhlee** [comment]
Thank you for an important comment...! I did not expect "image_load()" to not work properly.

**Comment by juhlee** [comment]
When first opening the notebook, did the two default sample images (cityscape.jpg, sunset.jpg) not load properly?

**Comment by MichielStock** [comment]
no, check it by loading the hyperlink to the notebook in Pluto, this is how people will use it.

**Comment by juhlee** [comment]
Thank you!

- I used "try/catch" statement to first try loading locally, then download the image if an local loading did not work.
- I then confirmed the notebook to run fine in another colleague's (outside of STMO course) computer.

**Comment by vandeveldet** [comment]
Great notebook and very clear walkthrough. Nice to have options available and see the effect on the clustering/color transfer. Initial issues with package loading were resolved by an update by @juhlee. Feel free to add a function when taking a repetitive action. Some small typo's: Eucl_i_dian distance and function reassi_gn_ing(). It worked well when adding new figures for Patrick and Spongebob ;).
patrick
